### PR TITLE
Don't upload node_modules to App Service

### LINKED ISSUE #540: Don't upload node_modules to App Service
# Motivation

We use `npm` to build the static code for the frontend, but once the static files are built, we no longer need `node_modules`. By uploading it to App Service, we are making the package much larger than it needs to be.

# How would you feel if this feature request was implemented?

🤏

# Requirements

A list of requirements to consider this feature delivered
- Remove `node_modules` from deployment package

# Tasks

_To be filled in by the engineer picking up the issue_

### PR DESCRIPTION
**Pull request by ross-p-smith**
## Purpose
Fixes #540

When we package the `web` service we zip the whole `code` folder which includes tests and node modules.

Below are the differences in zip files
```text
-rw-r--r-- 1 vscode vscode 76694284 Mar 27 21:17 /tmp/chat-with-your-data-solution-accelerator-web-azddeploy-1711574224.zip
-rw-r--r-- 1 vscode vscode  2313542 Mar 27 21:29 /tmp/chat-with-your-data-solution-accelerator-web-azddeploy-1711574955.zip
```

What we do now is 
1) Create a new `dist` folder - which is git ignored
2) Amend the npm build to output to this folder
3) Amend poetry to dump the requirements to this folder
4) Copy just the code that we want to this folder
5) For good measure don't copy the tests when we use docker too!

## How to Test
Tested manually
image

### CODE REVIEW

File: azure.yaml [hunk 1]
```diff
@@ -9,16 +9,17 @@ services:
     project: ./code
     language: py
     host: appservice
+    dist: ./dist
     hooks:
       prepackage:
         windows:
           shell: pwsh
-          run: poetry export -o requirements.txt;cd ./frontend;npm install;npm run build;
+          run: cd frontend;npm install;npm run build;cd ..;poetry export -o dist\requirements.txt;cp app.py dist;cp -r backend dist
```
Comment (cecheta) [inline azure.yaml:17]: Just checking this works in PowerShell?

File: azure.yaml [hunk 2]
```diff
@@ -9,16 +9,17 @@ services:
...
           interactive: true
           continueOnError: false
         posix:
           shell: sh
-          run: poetry export -o requirements.txt;cd ./frontend;npm install;npm run build;
+          run: cd frontend;npm install;npm run build;cd ..;poetry export -o dist/requirements.txt;cp app.py dist;cp -r backend dist
```
Comment (cecheta) [inline azure.yaml:22]: These are getting a bit long, thoughts on putting them in their own scripts?

File: azure.yaml (hunk 1 above)
Comment (ross-p-smith) [reply azure.yaml]: Yes it does!

File: azure.yaml (hunk 2 above)
Comment (ross-p-smith) [reply azure.yaml]: Now in a script

**Review by cecheta — APPROVED** [review]
Nice :)

**Pull request merged**